src/clr-core/button/button.element.ts

```diff
@@ -5,18 +5,216 @@
  */
 
 import { registerElementSafely } from '@clr/core/common';
-import { html, LitElement } from 'lit-element';
-
+import {
+  CwcBaseButton,
+  getElementWidthUnless,
+  getTranslateForChromeRenderingBugUnless,
+  hiddenButtonTemplate,
+  toggleDisabledAttribute,
+} from '@clr/core/common';
+import { html, property } from 'lit-element';
 import { styles } from './button.element.css';
 
+export enum ClrLoadingState {
+  DEFAULT,
+  LOADING,
+  SUCCESS,
+  ERROR,
+}
+
+/**
+ * @noInheritDoc
+ * @element `cwc-button`
+ * @slot `default` - Content slot for inside the button
+ * @styleAttr `action` [default|outline|link]
+ * @styleAttr `state` [default|primary|inverse|success|warning|danger]
+ * @styleAttr `size` - [small|default]
+ * @cssProp `--clr-btn-vertical-margin`
+ * @cssProp `--clr-btn-horizontal-margin`
+ * @cssProp `--clr-btn-horizontal-padding`
+ * @cssProp `--clr-btn-vertical-padding`
+ * @cssProp `--clr-btn-padding`
+ * @cssProp `--clr-btn-height`
```

Comment:
So the doc gen update is now on master. To auto generate the docs the syntax had to be changed a bit.

```
@element cwc-button
@slot default - Content slot for inside the button
@attr action [default|outline|link]
@cssprop --clr-btn-vertical-margin
```

No backticks. `styleAttr` is now `attr`and `@cssProp` is `@cssprop`. When you run the full build it will generate a `cwc-button.md` file which now can be source controlled. This change is not a PR blocker as I can update it as well once merged.